### ISSUE TITLE
[NBug] Could not find a part of the path 'k:\git_repos\myrepo.git\...

### ISSUE THREAD

**Issue by kolomenkin** [gitextensions/gitextensions#8647]
:warning: The sections below must be filled in and this text must be removed or the issue will be closed.

## Current behaviour

Application crashes during closing the window.

## Expected behaviour

I expect window to be closed without a crash inside of app.

## Steps to reproduce

1. Create working dir with external git database on a separate disk. .git file with path inside is used instead of .git dir:
========= .git content:  =============
gitdir: k:/git_repos/myrepo.git
=============================

2. Unmount the drive K: (flash disk, network drive, crypto drive, mounted NTFS folder ...)

3. Run GitExtensions by selecting "Commit" in working folder's context menu.

4. Windows will not be able to show the differences or to commit anything. Try closing the window by clicking right-top close button.

## Error Details
```
System.IO.DirectoryNotFoundException: Could not find a part of the path 'k:\git_repos\myrepo.git\GitExtensions.amend'.
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.File.InternalDelete(String path, Boolean checkHost)
   at System.IO.Abstractions.FileWrapper.Delete(String path)
   at GitCommands.CommitMessageManager.set_AmendState(Boolean value)
   at GitUI.CommandsDialogs.FormCommit.OnFormClosing(FormClosingEventArgs e)
   at System.Windows.Forms.Form.CheckCloseDialog(Boolean closingOnly)
```

## Additional information

This is not normal situation for your app. But it should not be enough for application to crash.

## Environment
- Git Extensions 3.3.1.7897
- Build 5a97671645532bcedc443bac7b727f40db47cb5c
- Git 2.26.0.windows.1
- Microsoft Windows NT 10.0.19042.0
- .NET Framework 4.8.4250.0
- DPI 96dpi (no scaling)

**Comment by kolomenkin** [comment]
From all these issues the following two have the same callstack:

https://github.com/gitextensions/gitextensions/issues/8353
https://github.com/gitextensions/gitextensions/issues/8232

But all previous guys had a bit different way to reproduce the problem.

**Comment by gerhardol** [comment]
There are related changes in #7739, included in 3.4. Please upgrade.

**Comment by RussKie** [comment]
Please update to the latest. 
Please ping us if the problem persists and we'll re-open it.
